src/tools/compiletest/src/runtest.rs

```diff
@@ -2608,13 +2608,14 @@ impl<'test> TestCx<'test> {
         // Because we use custom prefixes, we also have to register the default prefix.
         filecheck.arg("--check-prefix=CHECK");
 
-        // Some tests use the current revision name as a check prefix.
+        // The current revision name can also be used as a check prefix
         if let Some(rev) = self.revision {
-            filecheck.arg("--check-prefix").arg(rev);
+            filecheck.arg(format!("--check-prefix=CHECK-{}", rev.to_uppercase()));
```

Comment:
I'm not sure if it's appropriate to add `CHECK-` to all prefixes. For me, uppercase is enough to emphasize that this is not an ordinary comment.

Reply:
Hmm, I don't mind using `CHECK-` for everything either, but since it's a hidden convention, we at least need documentation to explain it.

Reply:
I did update the in-repo docs as part of this https://github.com/rust-lang/rust/pull/128018/files#diff-b40381314bcd0d5dd401af44ac13150c8129d97c97c6a76cbae9e3841a551329 :) and I would follow this with a dev guide update